editoast/src/schema/rolling_stock/mod.rs

```diff
@@ -0,0 +1,109 @@
+pub mod light_rolling_stock;
+
+use crate::schema::rolling_stock_livery::RollingStockLiveryMetadata;
+use diesel_json::Json as DieselJson;
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+use std::collections::HashMap;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct RollingStock {
+    pub id: i64,
+    pub name: String,
+    pub version: String,
+    pub effort_curves: DieselJson<EffortCurves>,
+    pub base_power_class: String,
+    pub length: f64,
+    pub max_speed: f64,
+    pub startup_time: f64,
+    pub startup_acceleration: f64,
+    pub comfort_acceleration: f64,
+    pub gamma: DieselJson<Gamma>,
+    pub inertia_coefficient: f64,
+    pub features: Vec<String>,
+    pub mass: f64,
+    pub rolling_resistance: DieselJson<RollingResistance>,
+    pub loading_gauge: String,
+    pub metadata: DieselJson<RollingStockMetadata>,
+    pub power_restrictions: Option<JsonValue>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct RollingStockWithLiveries {
+    #[serde(flatten)]
+    pub rolling_stock: RollingStock,
+    pub liveries: Vec<RollingStockLiveryMetadata>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Gamma {
+    #[serde(rename = "type")]
+    gamma_type: String,
+    value: f64,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+#[allow(non_snake_case)]
+pub struct RollingResistance {
+    #[serde(rename = "type")]
+    rolling_resistance_type: String,
+    A: f64,
+    B: f64,
+    C: f64,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct RollingStockMetadata {
+    detail: String,
+    family: String,
+    #[serde(rename = "type")]
+    rolling_stock_type: String,
+    grouping: String,
+    series: String,
+    subseries: String,
+    unit: String,
+    number: String,
+    reference: String,
+}
+
+// Effort curves schema
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct EffortCurveConditions {
+    comfort: Option<String>, // enum ConfortType
```

Comment:
We can define an enum `ConfortType` and use it here.

Reply:
Done !